### PR TITLE
Adding documentation for all features

### PR DESCRIPTION
**Pull request by Nyholm**
This could be helpful to quickly understand what Carson is doing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,3 +25,84 @@ are up to date.
...
+
+## Features
+
+Here is all the things Carson can help you with. Not all features are enabled. A
+list of enabled features for a specific repository is defined in `config/services.yaml`.
+
+### Automatic labeling from content
+
+Using the issue template, carson can figure out to add labels "Bug", "Feature",
+"BC Break" and "Deprecation" on an issue and pull request.
```
Comment (wouterj) [inline README.md:37]:
```suggestion
Using the issue template, Carson adds labels "Bug", "Feature", "BC Break"
and "Deprecation" on issues and pull requests.
```

File: README.md [hunk 2]
```diff
@@ -25,3 +25,84 @@ are up to date.
...
+When a PR is opened, then Carson will add "Needs Review".
+
+### Add "Needs Review" label on bugs issues
+
+If an issue is labeled with "Bug", then Carson will add "Needs Review".
+
+### Update pull request title with component labels
+
+When a PR is labeled, we will look for component labels, ie labels with color #dddddd.
+These labels' names will be added to the PR title.
```
Comment (wouterj) [inline README.md:50]:
```suggestion
When a PR is labeled, Carson looks for component labels (i.e. labels with color #dddddd).
These label's names will be added to the PR title (e.g. `[HttpKernel]`).
```

File: README.md [hunk 3]
```diff
@@ -25,3 +25,84 @@ are up to date.
...
+
+### Close draft pull requests
+
+When a PR is open as "draft", we add a comment to encourage people to mark it as
+"ready to review". If no action is taken, Carson will close the PR.
```
Comment (wouterj) [inline README.md:55]:
```suggestion
When a PR is open as "draft", Carson adds a comment to encourage people to mark it as
"ready to review". If no action is taken, Carson will close the PR in 2 hours.
```

(not sure about the nr of hours)

File: README.md [hunk 4]
```diff
@@ -25,3 +25,84 @@ are up to date.
...
+
+### Add milestone to PRs
+
+When a new PR is open and it does not target the default branch, then Carson will
+update the milestone of the PR to a existing milestone that matches the target branch.
```
Comment (wouterj) [inline README.md:60]:
```suggestion
When a new PR is opened and it does not target the default branch or current version, then
Carson will update the milestone of the PR to a existing milestone that matches the target branch.
```

File: README.md [hunk 5]
```diff
@@ -25,3 +25,84 @@ are up to date.
...
+
+The process can be interrupted with anyone making a comment on the issue or the
+"Keep open" label is added.
+
+### Suggest reviewers
+
+Carson can look at the PR to figure out who will be a good candidate to review the
+changes. This can be triggered by someone can make a comment in a PR to say
+@carsonbot find me a reviewer please" (or really anything with "@carsonbot" and
+"review" on the same line).
```
Comment (wouterj) [inline README.md:93]:
```suggestion
`@carsonbot find me a reviewer please` (or really anything with `@carsonbot` and
`review` on the same line).
```

File: README.md [hunk 6]
```diff
@@ -25,3 +25,84 @@ are up to date.
...
+
+If a PR is opened towards a branch that is not maintained anymore, Carson will
+kindly explain to the author what to do.
+
+### Open issues when docs for config reference is incomplete
+
+The Symfony documentation includes some pages with "configuration reference", to
+make sure these are always up to date, Carson will look at the bundles' defined
+config and compare that to what is documented. If Carson finds something that is
+missing, it will open an issue to the symfony/symfony-docs.
```
Comment (wouterj) [inline README.md:108]:
```suggestion
missing, it will open an issue in the symfony/symfony-docs repository.
```

**Review by wouterj — APPROVED** [review]
Good idea! Sorry for the many small suggestions

File: README.md (hunk 2 above)
Comment (Nyholm) [reply README.md:50]: I think it should be `These labels' names`. I'm referring to many `labels` and if they are owning something I should add the `'s`. But writing `labels's` is something that should be avoided. One will short it to `labels'`.

Comment (Nyholm) [reply README.md]: I updated your comment to get the other changes.

**Comment by Nyholm** [comment]
Thank you for the review. But I cannot take credit for the idea. 

Ryan suggested this to me.

**Comment by weaverryan** [comment]
This makes me very happy. Thanks Tobias!

**Comment by Nyholm** [comment]
Thank you for merging

**Pull request merged**